test/test_core.py

```diff
@@ -9493,14 +9493,21 @@ def test_pthread_dylink_main_module_1(self):
   def test_Module_dynamicLibraries(self, args):
     # test that Module.dynamicLibraries works with pthreads
     self.emcc_args += args
-    self.emcc_args += ['--pre-js', 'pre.js']
+    self.emcc_args += ['--extern-post-js=extern-post.js']
+    self.set_setting('MODULARIZE')
+    self.set_setting('EXPORT_NAME=ModuleFactory')
```

Comment:
I don't think we need these two settings to very that `dynamicLibraries` is shared do we?